cognite_toolkit/_cdf_tk/loaders/_resource_loaders.py

```diff
@@ -2254,15 +2265,15 @@ def load_resource(
             resources = [resources]
 
         for resource in resources:
-            try:
-                if config := resource.get("config", None):
-                    resource["config"] = yaml.dump(config, indent=4)
-            except Exception:
-                print(
-                    f"[yellow]WARNING:[/] configuration for {resource.get('external_id')} could not be parsed as valid YAML, which is the recommended format.\n"
-                )
-                resource["config"] = resource.get("config", None)
-
+            config_raw = resource.get("config")
+            if isinstance(config_raw, (dict, list)):
+                try:
+                    resource["config"] = yaml.safe_dump(config_raw, indent=4)
+                except yaml.YAMLError as e:
+                    print(
+                        f"[yellow]WARNING:[/] configuration for {resource.get('external_id', 'missing')} could not be parsed "
+                        f"as valid YAML, which is the recommended format. Error: {e}\n"
+                    )
```

Comment:
Warning class?

Reply:
That is not an easy fix in this context.